File: backend/app/schemas/health_metric.py
from typing import Optional, List
from pydantic import BaseModel, Field, ConfigDict

class BloodGlucoseCreate(BaseModel):
    patient_id: str = Field(default="mock-elderly-user-123", json_schema_extra={"example": "mock-elderly-user-123"})
    glucose_level: float = Field(..., ge=20.0, le=600.0, json_schema_extra={"example": 145.0})  # mg/dL
    measurement_type: str = Field(default="Fasting", json_schema_extra={"example": "Fasting"})  # Fasting, Pre-meal, Post-meal, Bedtime, Random
    notes: Optional[str] = Field(None, json_schema_extra={"example": "Felt a bit dizzy before breakfast"})

class BloodGlucoseResponse(BaseModel):
    id: str
    patient_id: str
    glucose_level: float
    measurement_type: str
    status: str  # "Normal", "Hypoglycemia", "Hyperglycemia"
    safety_recommendation: str
    audio_warning_script: str
    caregiver_alert_sent: bool
    created_at: str
    notes: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)

class HealthSummaryResponse(BaseModel):
    patient_id: str
    total_logs: int
    average_glucose: float
    latest_reading: Optional[BloodGlucoseResponse] = None
    hypoglycemia_count: int
    hyperglycemia_count: int

File: frontend/lib/models/health_metric.dart
class BloodGlucoseLog {
  final String id;
  final String patientId;
  final double glucoseLevel;
  final String measurementType;
  final String status;
  final String safetyRecommendation;
  final String audioWarningScript;
  final bool caregiverAlertSent;
  final String createdAt;
  final String? notes;

  BloodGlucoseLog({
    required this.id,
    required this.patientId,
    required this.glucoseLevel,
    required this.measurementType,
    required this.status,
    required this.safetyRecommendation,
    required this.audioWarningScript,
    required this.caregiverAlertSent,
    required this.createdAt,
    this.notes,
  });

  factory BloodGlucoseLog.fromJson(Map<String, dynamic> json) {
    return BloodGlucoseLog(
      id: json['id'] ?? '',
      patientId: json['patient_id'] ?? '',
      glucoseLevel: (json['glucose_level'] as num).toDouble(),
      measurementType: json['measurement_type'] ?? 'Fasting',
      status: json['status'] ?? 'Normal',
      safetyRecommendation: json['safety_recommendation'] ?? '',
      audioWarningScript: json['audio_warning_script'] ?? '',
      caregiverAlertSent: json['caregiver_alert_sent'] ?? false,
      createdAt: json['created_at'] ?? '',
      notes: json['notes'],
    );
  }
}
